### PR TITLE
Week 1 assessment heather

### PR DESCRIPTION
**Pull request by hroulston**

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -14,14 +14,53 @@ var temp1 = 35
...
+}
+console.log(singleArray(myNumbers1, myNumbers2))
+
+// Next the function needs to sort all of the numbers that are in that single array. So I will console,log what I have above and include the .sort() method. My issue however, is I am pretty sure .sort() works to alphabetize items and we are working with numbers. So let's try it, see what happens and adjust from there.
+
+// console.log(singleArray(myNumbers1, myNumbers2).sort())
+
+ // And the result was the numbers were all wonky. I did some research and found the comparison function (a,b => a-b) which takes two values and subtracts the first from the second. If the answer is negative the second value is bigger and if the answer is positive the first value is bigger. As the function runs through the numbers it will move them accordingly. If I use this comparison function and pass it into the sort function it should sort the values in ascending order.
+
+ console.log(singleArray(myNumbers1, myNumbers2).sort((a,b)=>(a-b)))
```
Comment (WalkerTexas-coder) [inline code-challenges.js:61]: This is excellent logic! The next step is to bring this method into the function so the logic is all happening in one place. Good stuff

File: code-challenges.js [hunk 2]
```diff
@@ -14,14 +14,53 @@ var temp1 = 35
...
+
+ // And bam, it worked. So ecxiting.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:63]: this made me smile ^_^

File: code-challenges.js [hunk 3]
```diff
@@ -30,7 +69,47 @@ var myNumbers2 = [8, -7, 22, 9, 13]
...
+// }
+
+// The function then needs to take a string and return the string with the letters in reverse.
+
+// const reverse = (string)=> {
+// return string.reverse()
+// }
+// console.log(reverse(myString1))
+
+// I tried to pass a variable in as an argument and return that variable with .reverse() but that threw an error. I apparently need to really work on what methods work with strings and what works with arrays :/
```
Comment (WalkerTexas-coder) [inline code-challenges.js:84]: Good catch!

File: code-challenges.js [hunk 4]
```diff
@@ -30,7 +69,47 @@ var myNumbers2 = [8, -7, 22, 9, 13]
...
+  var reverseArray = splitStr.reverse()
+  var finalStr = reverseArray.join("")
+  return finalStr
+ }
+
+ // fingers crossed...
+
+ console.log(reverse(myString2))
+
+ // alright, it worked!
```
Comment (WalkerTexas-coder) [inline code-challenges.js:112]: That was so fun! I read that out loud, it was a joy ride!

So this style of Pseudo coding you're doing where you comment then write the code beneath is my ABSOLUTE favorite! I appreciate you working through each step as it's own problem then bringing it all togther!

File: code-challenges.js [hunk 5]
```diff
@@ -41,7 +120,39 @@ var numberExample2 = 27
...
+const subtraction = (num1, num2) =>{
+  if(num1 === num2){
+    return "numbers are equal"
+  }else if(num1 < num2){
+    return num2 - num1
+  }else{
+    return num1 - num2
+  }
+}
+console.log(subtraction(numberExample3, numberExample4 ))
```
Comment (WalkerTexas-coder) [inline code-challenges.js:155]: Crushed it!

File: code-challenges.js [hunk 6]
```diff
@@ -52,3 +163,67 @@ var numberExampleRefactor2 = "hello"
...
+const subtraction = (num1, num2) =>{
+  if(num1 === num2){
+    return "numbers are equal"
+  }else if(num1 < num2){
+    return num2 - num1
+  }else if(num1 > num2){
+    return num1 - num2
+    // Refactor the code to inform the user if the variable passed through is not a number
+  }else{
+    return "Your input is not a number"
```
Comment (WalkerTexas-coder) [inline code-challenges.js:178]: simple, clean, perfect!

File: code-challenges.js [hunk 7]
```diff
@@ -52,3 +163,67 @@ var numberExampleRefactor2 = "hello"
...
+         return "numbers are equal"
+      }else if(num1 < num2){
+         return num2 - num1
+      }else{
+         return num1 - num2
+       }
+     }
+  console.log(subtraction(numberExampleRefactor1, numberExampleRefactor2))
+
+// Yay!!! I did have a few runs of syntatical errors because of extra {} from my copy pasting. However, I am still celebrating.
```
Comment (WalkerTexas-coder) [inline code-challenges.js:229]: This is excellent work and totally worthy of a happy dance!

File: coding-concepts.js [hunk 8]
```diff
@@ -29,23 +29,24 @@ var languages = ["JavaScript", "Ruby", "Python", "C++"]
...
 // --------------------4) What will this log?
 
 var weekendDays = ["saturday", "sunday"]
 // console.log(weekendDays.toUpperCase())
 
-// a) Your answer:
-// b) Verify and explain:
+
+// a) Your answer: ["SATURDAY", "SUNDAY"]
+// b) Verify and explain: TypeError weekendDays.toUpperCase is not a function. This threw me for a quick loop and then I realized when I reasearched that .toUpperCase works on strings. weekendDays is assigned an array, not a string. So it will throw an error.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:43]: Good catch!

File: coding-concepts.js [hunk 9]
```diff
@@ -29,23 +29,24 @@ var languages = ["JavaScript", "Ruby", "Python", "C++"]
...
 
 // --------------------5) What will this log?
 
 var dataTypes = ["number", "string", "Boolean"]
 // console.log(typeof dataTypes.length)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: String
+// b) Verify and explain: Number. Hmmm. I am assuming it just returned the first value in dataTypes. When I did some research I read that typeof works on primitive data types and when given anything besides the 6 data types we learned plus bignit and function it will return object. When I ran the code with out .length it returned object. So I can only conclude type of will not work on the variable becuase it holds an array (which is not a primitive data type) and .length was added to throw us off into thinking it would go through the whole array.
```
Comment (WalkerTexas-coder) [inline coding-concepts.js:52]: So number here is becuase the .length property is telling us how long the array is 3. So the datatype of 3 is number and typeof is running on 3

ala, console.log(typeof 3)

File: interview-questions.md [hunk 10]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a small coding machine that runs a block of code to create a result when called upon.
 
-  Researched answer:
+  Researched answer: Functions are self contained bits of code that accomplish a specific task. They take in data and then return a result. One of its best features is its reusability; this not only saves time and energy but produces cleaner more concise code.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:12]: Both answers are great!

File: interview-questions.md [hunk 11]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. In regards to functions, what is the difference between an argument and a parameter?
 
-  Your answer:
+  Your answer: A parameter is the place holder in a function indicated by parenthesis. An argument is what you put within the parenthesis when you call upon the function.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:18]: Great job!

File: interview-questions.md [hunk 12]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: An index is the place of an item. The value is what is at the index. For example: var hello = "hello". The value is "h" and its position is index 0.
 
-  Researched answer:
+  Researched answer: One of the best descriptions I found was as an example that an array is made of different boxes. The index is the number of the box we are interested in. What you find inside the box is the value.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:28]: Awesome, would love for you to share this with the class!

File: interview-questions.md [hunk 13]
```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: When pair programming there is always a driver and a navigator. The driver is the person who is typing out the code and the navigator is telling the driver what to type while reviewing the code. In order for a pair programming session to be successful communication is essential. Both parties should be clear and specific as well as open to trying new ideas. Switching roles often is highly encouraged.
 
-  Researched answer:
+  Researched answer: Pair programming is when two programmers work together on a single project. There is a driver and a navigator (or observer). The navigator reviews as well as thinks about what direction they should be working in to achieve their goal. The driver should be able to focus on their task and rely on the navigator as their guide and second pair of eyes. This type of programming typically results in fewer defects.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:44]: Both answers are great!

**Review by WalkerTexas-coder — APPROVED** [review]
Heather this was so fun to go over! Great work and AMAZING pseudo code. Your logic is super clear. I can watch as develop and refine your skills as you go through the assessment somethign that is quite cool to see. The only feedback I have is try to get all the logic for a given challenge into one function-- but thats not a big deal week one-- just a goal. Honestly I am looking forawrd to next weeks assessment. Great great work! Your PR is approved!

File: coding-concepts.js (hunk 9 above)
Comment (hroulston) [reply coding-concepts.js:52]: Yep realized this when we went over it in class that I completely (and confidently I might add) totally missed the mark. I always have to tell myself to stop over thinking it when it comes to javascript.

**Pull request merged**